// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// rank.proto
syntax = "proto3";
import "user_info.proto";
import "item_info.proto";

message RankRequest {
  string log_id = 1;
  user_info.UserInfo user_info = 2;
  repeated item_info.ItemInfo item_infos = 3;
}

message RankResponse {
    message Error {
        uint32 code = 1;
        string text = 2;
    }
    message ScorePair {
        string nid = 1;
        float score = 2;
    };
    Error error = 1;
    repeated ScorePair score_pairs = 2;
};

service RankService {
    rpc rank_predict(RankRequest) returns (RankResponse);
};
